src/main/java/com/zero_x_baadf00d/play/module/redis/cache/AsyncCacheRedisImpl.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016 - 2017 Thibault Meyer
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.zero_x_baadf00d.play.module.redis.cache;
+
+import akka.Done;
+import com.zero_x_baadf00d.play.module.redis.PlayRedis;
+import play.cache.AsyncCacheApi;
+import play.cache.SyncCacheApi;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static java.util.concurrent.CompletableFuture.supplyAsync;
+
+/**
+ * Implementation of {@code PlayRedis} as Async Cache Api.
+ *
+ * @author Felipe Bonezi
+ * @version 20.11
+ * @see PlayRedis
+ * @since 20.11.05
+ */
+@Singleton
+public class AsyncCacheRedisImpl implements AsyncCacheApi {
+
+    /**
+     * Keep all keys saved into cache.
+     *
+     * @since 20.11.05
+     */
+    private final Set<String> keys = new HashSet<>(100);
```

Comment:
I'm not sure. This mechanism will not work if several instances of the application are running. The simplest would be, in my opinion, to use the `getConnection` method to execute the `KEYS` method which allows to find all the keys matching needed patter.

You can use a hardcoded prefix or allow user to change it in the configuration file. By example if you use the prefix `cache.`, the method `removeAll` will use Redis connection to find all keys matching `cache.*` and delete them.

Redis doc : https://redis.io/commands/keys


**Example**
```java
@Override
public CompletionStage<Done> removeAll() {
   return supplyAsync(() -> {
        this.playRedis.getConnection(0).keys("search pattern").forEach(this::remove);
        return Done.done();
    });
}
```